### PR TITLE
config: allow four byte-size config to be specified using human-readable units ("100 GiB")

### PR DESCRIPTION
**Pull request by kennytm**
### What problem does this PR solve? 

Several config specifies a large byte size, which is not very readable (and some DBAs are still unfamiliar with the underscore-as-number-separator TOML feature).

### What is changed and how it works?

Using [`docker/go-units`](https://pkg.go.dev/github.com/docker/go-units) (already used by Dumpling and PD), we support parsing human-readable byte size for the following config:

* tikv-importer.region-split-size (100_663_296 → '96 MiB')
* mydumper.read-block-size (65536 → '64 KiB')
* mydumper.batch-size (107_374_182_400 → '100 GiB')
* mydumper.max-region-size (268_435_456 → '256 MiB')

Byte size can be specified using integers (`65536` or `65_536`, backward-compatible with existing config), floating point (`6.5536e+4`), or a string involving byte units (`'64k'`, `'64 K'`, `'64KiB'`, `'64 kb'`, `'0.0625 MB'`, all equivalent).

(This PR is submitted mainly because we expect that disk-quota is also going to need some huge numbers in a custom config, similar to TiKV's `ReadableSize` struct. Interestingly no such thing exists in TiDB except in dealing with INSPECTION_RULES.)

### Check List 

Tests 

 - Unit test

Side effects

Related changes

 - Need to update the documentation
 - Need to be included in the release note

# Release notes

* The 4 configuration specified above can now accept human-readable format in the form `"2.5 GiB"`.

### CODE REVIEW

File: lightning/restore/checksum_test.go [hunk 1]
```diff
@@ -271,16 +271,17 @@ func (s *checksumSuite) TestGcTTLManagerSingle(c *C) {
...
 	c.Assert(err, IsNil)
 
-	time.Sleep(6 * time.Second)
+	time.Sleep(6*time.Second + 10*time.Millisecond)
 
-	// after 11 seconds, must at least update 5 times
+	// after 6 seconds, must at least update 5 times
 	val := atomic.LoadInt32(&pdClient.count)
-	c.Assert(val >= 5, IsTrue)
+	c.Assert(val, Ge, 5)
```
Comment (glorv) [inline lightning/restore/checksum_test.go:278]: This should be `GreaterEqual` ?

File: lightning/config/bytesize.go [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+	res, err := units.RAMInBytes(string(b))
+	if err != nil {
+		return err
+	}
+	*size = ByteSize(res)
+	return nil
+}
+
+// UnmarshalJSON implements json.Unmarshaler (for testing)
+func (size *ByteSize) UnmarshalJSON(b []byte) error {
```
Comment (glorv) [inline lightning/config/bytesize.go:37]: shall we also implement `MarshalText`and `MarshalJSON`

Comment (kennytm) [reply lightning/config/bytesize.go:37]: Seems no need to implement the Marshal methods as the unit test passes without them.

File: lightning/restore/checksum_test.go (hunk 1 above)
Comment (kennytm) [reply lightning/restore/checksum_test.go:278]: thanks, fixed.

**Review by glorv — APPROVED** [review]
LGTM

**Comment by overvenus** [comment]
LGTM

**Pull request merged**